data_steward/tools/delete_stale_test_datasets.py

```diff
@@ -0,0 +1,103 @@
+"""
+Delete stale datasets in test environment. 
+"""
+from datetime import datetime, timezone
+import logging
+import os
+
+# Project imports
+from utils import bq, pipeline_logging
+
+LOGGER = logging.getLogger(__name__)
+
+
+def _check_project(bq_client):
+    """
+    Check if the project is set to test.
+
+    :param bq_client: Client
+    :return: None if the project is set properly
+    :raise: ValueError if project is not 'aou-res-curation-test'
+    """
+
+    if 'test' not in bq_client.project:
+        raise ValueError(
+            f'Wrong project: {bq_client.project}. This script runs only for test environment'
+        )
+
+    return None
+
+
+def _filter_stale_datasets(bq_client, first_n: int = None):
+    """
+    Get the first n datasets that meet all of the following criteria:
+    1. Datasets older than 90 days
+    2. Empty datasets
+
+    :param bq_client: Client
+    :param first_n: number of datasets to return. If not specified, return all.
+    :return: list of dataset names that are stale
+    """
+
+    stale_datasets, n = [], 0
+    now = datetime.now(timezone.utc)
+
+    dataset_names = [
+        dataset.dataset_id for dataset in bq_client.list_datasets()
+    ]
+
+    for dataset_name in dataset_names:
+
+        if first_n and n >= first_n:
+            break
+
+        dataset_created = bq_client.get_dataset(dataset_name).created
+
+        if (now - dataset_created).days <= 90:
+            continue
+
+        LOGGER.critical(
+            f"{dataset_name}, {type(bq_client.list_tables(dataset_name))}")
+        LOGGER.critical(
+            f"{dataset_name}, {dir(bq_client.list_tables(dataset_name))}")
+        LOGGER.critical(
+            f"{dataset_name}, {bq_client.list_tables(dataset_name).pages}")
+        LOGGER.critical(
+            f"{dataset_name}, {dir(bq_client.list_tables(dataset_name).item_to_value)}"
+        )
+        #if next(bq_client.list_tables(dataset_name), None):
+        #    continue
+        if list(bq_client.list_tables(dataset_name)) == 0:
```

Comment:
`bq_client.list_tables()` returns `HTTPIterator`. This is not an iterator. If used as iterator (used with `next()` for example), it throws `TypeError: 'HTTPIterator' object is not an iterator` with CircleCi execution. For some reason, when I run it locally, this error does not happen. Some settings in my environment seem to interpret `HTTPIterator` as an iterator.

To avoid `TypeError`, StackOverflow told me I can cast that to a list:
https://stackoverflow.com/questions/67427634/how-to-get-the-next-blob-in-python-google-cloud-storage-library-method-list-blob

The mentioned solution is used in the following script in our repository as well. 
https://github.com/all-of-us/curation/blob/ac9f38b2f4580ae806121dd929293159132c7d2a/data_steward/utils/gcs.py#L37

So, I am using `list(bq_client.list_tables())` as workaround here.

Related page in Google API doc:
https://googleapis.dev/python/google-api-core/latest/page_iterator.html